pandas/plotting/_matplotlib/core.py

```diff
@@ -423,13 +425,19 @@ def _compute_plot_data(self):
         if is_empty:
             raise TypeError("no numeric data to plot")
 
-        # GH25587: cast ExtensionArray of pandas (IntegerArray, etc.) to
-        # np.ndarray before plot.
-        numeric_data = numeric_data.copy()
-        for col in numeric_data:
-            numeric_data[col] = np.asarray(numeric_data[col])
+        def convert_to_ndarray(data):
```

Comment:
Consider moving ``convert_to_ndarray`` out of the scope of the function ``_compute_plot_data``.

Reply:
yes pls do this

Reply:
@jreback I added the changes to the latest commit. Can you please review it again ?